R/FilterState.R

```diff
@@ -46,26 +46,24 @@ FilterState <- R6::R6Class( # nolint
   public = list(
 
     #' @description
-    #' Initialize a `FilterState` object
-    #' @param x (`vector`)\cr
+    #' Initialize a `FilterState` object.
+    #' @param x (`vector`)
     #'   values of the variable used in filter
-    #' @param x_reactive (`reactive`)\cr
+    #' @param x_reactive (`reactive`)
     #'   returning vector of the same type as `x`. Is used to update
     #'   counts following the change in values of the filtered dataset.
     #'   If it is set to `reactive(NULL)` then counts based on filtered
     #'   dataset are not shown.
-    #' @param slice (`teal_slice`)\cr
+    #' @param slice (`teal_slice`)
     #'   object created by [teal_slice()]
-    #' @param extract_type (`character(0)`, `character(1)`)\cr
+    #' @param extract_type (`character`)
     #'   specifying whether condition calls should be prefixed by `dataname`. Possible values:
-    #' \itemize{
-    #' \item{`character(0)` (default)}{ `varname` in the condition call will not be prefixed}
-    #' \item{`"list"`}{ `varname` in the condition call will be returned as `<dataname>$<varname>`}
-    #' \item{`"matrix"`}{ `varname` in the condition call will be returned as `<dataname>[, <varname>]`}
-    #' }
+    #' - `character(0)` (default) `varname` in the condition call will not be prefixed
+    #' - `"list"` `varname` in the condition call will be returned as `<dataname>$<varname>`
+    #' - `"matrix"` `varname` in the condition call will be returned as `<dataname>[, <varname>]`
     #' @param ... additional arguments to be saved as a list in `private$extras` field
     #'
-    #' @return self invisibly
+    #' @return `self` invisibly
```

Comment:
@kartikeyakirar does `self` needs to be quoted with backticks? Like it was an object. What if you rewrite to something like `Object of class \code{FilterState} (invisibly).`

Reply:
:exclamation: `self` is not the same as object of class `FilterState`.

I have no opinion on quoting `self` ATM.

Reply:
I don't think `self` invisibly is enough in here. `Object of class \code{FilterState} (invisibly).` or at least `\code{FilterState} (invisibly)` to highlight the class of the returned value.

Reply:
`self` is more specific because it means _this very instance_ is returned, not just any `FilterState`. Needless to say, `self` implies the class rather unambiguously.

Reply:
In the context of R6 class documentation, the term self refers to the object itself, and it's common to use backticks to indicate that it's a specific term or object in the code. In r6 documentation also its quoted with backticks https://github.com/search?q=repo%3Ar-lib%2FR6%20self&type=code

Reply:
ok team, you convinced me my suggestion is not needed in here. resolving this one